src/ui/searchdialog.cc

```diff
@@ -103,11 +108,18 @@ qint64 SearchDialog::lastIndexOf(const data::BinData &pattern,
   return -1;
 }
 
-void SearchDialog::replace(qint64 pos, qint64 len, const data::BinData &data) {
-  // TODO: implement this
+void SearchDialog::replace(qint64 pos, const data::BinData &data) {
+  _hexEdit->setBytesValues(pos, data);
 }
 
-qint64 SearchDialog::findNext() {
+void SearchDialog::enableReplace(const QString& find, const QString& replace) {
+  ui->warning_label->setVisible(replace.length() && (replace.length() != find.length()));
```

Comment:
if you try to replace hex to utf8 then "6161" -> "bb" displays warning, but shouldn't